### ISSUE TITLE
extract_path errors when running controller tests

### ISSUE THREAD

**Issue by m1foley** [roidrage/lograge#168]
I see errors in `test.log` every time I call controller methods in my project's controller specs. The errors don't show up in development or production. The project is on Rails 4.2.5.1.

The specs are like:
```ruby
describe FooController, type: :controller do
  it "stuff" do
    post :foo, id: 1, format: "json"
  end
end
```

Error in `test.log`:
```
Could not log "process_action.action_controller" event. NoMethodError:
undefined method `index' for nil:NilClass
["/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/lograge-0.3.6/lib/lograge/log_subscriber.rb:55:in
`extract_path'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/lograge-0.3.6/lib/lograge/log_subscriber.rb:46:in
`initial_data'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/lograge-0.3.6/lib/lograge/log_subscriber.rb:35:in
`extract_request'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/lograge-0.3.6/lib/lograge/log_subscriber.rb:12:in
`process_action'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/subscriber.rb:100:in
`finish'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/log_subscriber.rb:83:in
`finish'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications/fanout.rb:102:in
`finish'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications/fanout.rb:46:in
`block in finish'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications/fanout.rb:46:in
`each'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications/fanout.rb:46:in
`finish'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications/instrumenter.rb:36:in
`finish'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications/instrumenter.rb:25:in
`instrument'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activesupport-4.2.5.1/lib/active_support/notifications.rb:164:in
`instrument'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionpack-4.2.5.1/lib/action_controller/metal/instrumentation.rb:30:in
`process_action'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionpack-4.2.5.1/lib/action_controller/metal/params_wrapper.rb:250:in
`process_action'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/activerecord-4.2.5.1/lib/active_record/railties/controller_runtime.rb:18:in
`process_action'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionpack-4.2.5.1/lib/abstract_controller/base.rb:137:in
`process'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionview-4.2.5.1/lib/action_view/rendering.rb:30:in
`process'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionpack-4.2.5.1/lib/action_controller/test_case.rb:639:in
`process'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionpack-4.2.5.1/lib/action_controller/test_case.rb:67:in
`process'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/actionpack-4.2.5.1/lib/action_controller/test_case.rb:520:in
`post'", "/Users/baz/foo/bar/spec/controllers/fun_controller_spec.rb:85:in
`block (3 levels) in <top (required)>'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:236:in
`instance_exec'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:236:in
`block in run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:477:in
`block in with_around_and_singleton_context_hooks'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:434:in
`block in with_around_example_hooks'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:478:in
`block in run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:618:in
`block in run_around_example_hooks_for'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:319:in
`call'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:319:in
`call'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-rails-3.4.0/lib/rspec/rails/example/controller_example_group.rb:191:in
`block (2 levels) in <module:ControllerExampleGroup>'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:424:in
`instance_exec'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:424:in
`instance_exec'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:389:in
`execute_with'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:620:in
`block (2 levels) in run_around_example_hooks_for'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:319:in
`call'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:319:in
`call'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-rails-3.4.0/lib/rspec/rails/adapters.rb:127:in
`block (2 levels) in <module:MinitestLifecycleAdapter>'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:424:in
`instance_exec'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:424:in
`instance_exec'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:389:in
`execute_with'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:620:in
`block (2 levels) in run_around_example_hooks_for'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:319:in
`call'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:319:in
`call'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:621:in
`run_around_example_hooks_for'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/hooks.rb:478:in
`run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:434:in
`with_around_example_hooks'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:477:in
`with_around_and_singleton_context_hooks'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example.rb:233:in
`run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:581:in
`block in run_examples'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:577:in
`map'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:577:in
`run_examples'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:543:in
`run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:544:in
`block in run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:544:in
`map'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/example_group.rb:544:in
`run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:115:in
`block (3 levels) in run_specs'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:115:in
`map'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:115:in
`block (2 levels) in run_specs'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/configuration.rb:1680:in
`with_suite_hooks'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:114:in
`block in run_specs'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/reporter.rb:77:in
`report'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:113:in
`run_specs'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:89:in
`run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:73:in
`run'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/lib/rspec/core/runner.rb:41:in
`invoke'",
"/Users/baz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/rspec-core-3.4.1/exe/rspec:4:in
`<top (required)>'", "/Users/baz/.rbenv/versions/2.1.6/bin/rspec:23:in `load'",
"/Users/baz/.rbenv/versions/2.1.6/bin/rspec:23:in `<main>'"]
```

**Comment by benlovell** [comment]
Is this still an issue for you?

**Comment by m1foley** [comment]
I am not seeing this error any more. I don't know what fixed it -- one guess is my upgrade to Rails 4.2.5.1 to 4.2.5.2.

Closing issue.
